### DISCUSSION TITLE
Feature request: React 19

### DISCUSSION THREAD

**Discussion by MangoMarcus** [algolia/instantsearch#6220] [Ideas] [answered]
With React 19 now becoming a release candidate I wondered when we might expect it to be supported? It's required by NextJS 15, which is also now a release candidate and contains some caching changes I'm keen to take advantage of 😁 

Thanks

**Comment by Haroenv** [comment]
We have not had the opportunity to try it out in detail, but happy to hear your feedback when you try it out, as in theory there shouldn't be any breaking changes in code that we're using.

Have you tried it out yet?

**Comment by MangoMarcus** [comment]
@Haroenv Apologies for the appallingly delayed reply

Using the latest instantsearch-react package in a react19 app gives 'InstantSearch' cannot be used as a JSX component.

<img width="846" alt="Screenshot 2024-11-18 at 15 17 34" src="https://github.com/user-attachments/assets/08bf601c-7944-4b81-bd5d-43e1fb25c579">

I'm also using Next 15, although I'm not using react-instantsearch-nextjs

**Reply by vonkoff** [reply]
Same thing on my end

<img width="1023" alt="Screenshot 2025-01-03 at 9 02 23 PM" src="https://github.com/user-attachments/assets/bf45ef9f-2055-4d3b-9c6b-fc9b4955e2ca" />

**Reply by vildantursic** [reply]
+1

**Comment by Haroenv** [comment, marked as answer]
This is fixed with versions:

```
"react-instantsearch": "7.14",
"react-instantsearch-nextjs": "0.4.0",
```

**Reply by anjitcubit** [reply]
just when I need it, `react-instantsearch-nextjs` fixed mine
